fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java

```diff
@@ -1752,7 +1764,11 @@ public PlanFragment visitPhysicalHashJoin(OptExpression optExpr, ExecPlan contex
                     context.getFragments().add(leftFragment);
 
                     leftFragment.mergeQueryGlobalDicts(rightFragment.getQueryGlobalDicts());
-                    estimateDopOfBroadcastJoinInPipeline(leftFragment);
+                    if (distributionMode.equals(HashJoinNode.DistributionMode.COLOCATE)) {
```

Comment:
can this condition expr exclude fragment instance contains HashJoinNode(shuffle)->HashJoinNode(bucket shuffle) ?

Reply:
@satanson No, `HashJoinNode(shuffle)->HashJoinNode(bucket shuffle)` will also set dop=1 and num_fragment_instance=CPUs/2.